File: FireSaverApi/Dtos/CompartmentDtos/CompartmentDto.cs
using System.Collections.Generic;
using FireSaverApi.DataContext;
using FireSaverApi.Dtos.IoTDtos;
using FireSaverApi.Dtos.TestDtos;

namespace FireSaverApi.Dtos.CompartmentDtos
{
    public class CompartmentDto
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public string SafetyRules { get; set; }
        public TestInputDto CompartmentTest { get; set; }
        public List<IotNewPositionDto> IoTs { get; set; }
    }
}

File: FireSaverSPA/src/app/Models/Compartment/compartmentDto.ts
import { IoTNewPostion } from "../IoTService/ioTNewPostion";
import { TestInput } from "../TestModels/testInput";
import { UserInfoDto } from "../UserService/userInfoDto";

export class CompartmentDto {
    id: number = 0;
    name: string;
    description: string;
    safetyRules: string;
    compartmentTest: TestInput;
    ioTs: IoTNewPostion[] = [];
    inboundUsers: UserInfoDto[] = []
}